go/vt/vtgate/buffer/flags.go

```diff
@@ -18,61 +18,80 @@ package buffer
+
+	fs.IntVar(&bufferDrainConcurrency, "buffer_drain_concurrency", 1, "Maximum number of requests retried simultaneously. More concurrency will increase the load on the PRIMARY vttablet when draining the buffer.")
+	fs.StringVar(&bufferKeyspaceShards, "buffer_keyspace_shards", "", "If not empty, limit buffering to these entries (comma separated). Entry format: keyspace or keyspace/shard. Requires --enable_buffer=true.")
+}
+
+func init() {
+	servenv.OnParseFor("vtgate", registerFlags)
+}
+
 func verifyFlags() error {
-	if *bufferWindow < 1*time.Second {
-		return fmt.Errorf("-buffer_window must be >= 1s (specified value: %v)", *bufferWindow)
+	if bufferWindow < 1*time.Second {
+		return fmt.Errorf("--buffer_window must be >= 1s (specified value: %v)", bufferWindow)
 	}
-	if *bufferWindow > *bufferMaxFailoverDuration {
-		return fmt.Errorf("-buffer_window must be <= -buffer_max_failover_duration: %v vs. %v", *bufferWindow, *bufferMaxFailoverDuration)
+	if bufferWindow > bufferMaxFailoverDuration {
+		return fmt.Errorf("--buffer_window must be <= --buffer_max_failover_duration: %v vs. %v", bufferWindow, bufferMaxFailoverDuration)
 	}
-	if *bufferSize < 1 {
-		return fmt.Errorf("-buffer_size must be >= 1 (specified value: %d)", *bufferSize)
+	if bufferSize < 1 {
+		return fmt.Errorf("--buffer_size must be >= 1 (specified value: %d)", bufferSize)
 	}
-	if *bufferMinTimeBetweenFailovers < *bufferMaxFailoverDuration*time.Duration(2) {
-		return fmt.Errorf("-buffer_min_time_between_failovers should be at least twice the length of -buffer_max_failover_duration: %v vs. %v", *bufferMinTimeBetweenFailovers, *bufferMaxFailoverDuration)
+	if bufferMinTimeBetweenFailovers < bufferMaxFailoverDuration*time.Duration(2) {
+		return fmt.Errorf("--buffer_min_time_between_failovers should be at least twice the length of --buffer_max_failover_duration: %v vs. %v", bufferMinTimeBetweenFailovers, bufferMaxFailoverDuration)
 	}
 
-	if *bufferDrainConcurrency < 1 {
-		return fmt.Errorf("-buffer_drain_concurrency must be >= 1 (specified value: %d)", *bufferDrainConcurrency)
+	if bufferDrainConcurrency < 1 {
+		return fmt.Errorf("--buffer_drain_concurrency must be >= 1 (specified value: %d)", bufferDrainConcurrency)
 	}
 
-	if *bufferKeyspaceShards != "" && !*bufferEnabled {
-		return fmt.Errorf("-buffer_keyspace_shards=%v also requires that -enable_buffer is set", *bufferKeyspaceShards)
+	if bufferKeyspaceShards != "" && !bufferEnabled {
+		return fmt.Errorf("--buffer_keyspace_shards=%v also requires that --enable_buffer is set", bufferKeyspaceShards)
 	}
-	if *bufferEnabled && *bufferEnabledDryRun && *bufferKeyspaceShards == "" {
+	if bufferEnabled && bufferEnabledDryRun && bufferKeyspaceShards == "" {
 		return errors.New("both the dry-run mode and actual buffering is enabled. To avoid ambiguity, keyspaces and shards for actual buffering must be explicitly listed in --buffer_keyspace_shards")
```

Comment:
'--' over here, is it because client still send us in '--'

Reply:
with the new way of doing things, it's always `--` for long flags, and `-` only for short flags.
See https://github.com/vitessio/enhancements/blob/main/veps/vep-4.md for details.